collect_app/src/main/java/org/odk/collect/android/tasks/GoogleSheetsAbstractUploader.java

```diff
@@ -565,117 +684,83 @@ else if (columnNames.contains(path)) { // if column present in sheet
-            String picasa_id = c.getString(c.getColumnIndex(Images.Media.PICASA_ID));
-            if (picasa_id == null || picasa_id.equalsIgnoreCase("")) {
-                // not in picasa, so continue
-            } else {
-                // got a picasa ID, make sure it exists in this
-                // particular album online
-                // if it does, go on
-                // if it doesn't, upload it and update the picasa_id
-                if (albumToUse.numPhotos != 0) {
-                    PicasaUrl photosUrl = new PicasaUrl(albumToUse.getFeedLink());
-                    AlbumFeed albumFeed = client.executeGetAlbumFeed(photosUrl);
-
-                    for (PhotoEntry photo : albumFeed.photos) {
-                        if (picasa_id.equals(photo.id)) {
-                            // already in picasa, no need to upload
-                            inPicasa = true;
-                            picasaPhoto = photo;
-                        }
-                    }
-                }
-            }
+    private String uploadFileToDrive(HashMap<String, String> photosToUpload, String key, com.google.api.services.drive.model.File folder, File toUpload) throws IOException {
 
-            // wasn't already there, so upload a new copy and update the
-            // content provder with its picasa_id
-            if (!inPicasa) {
-                String fileName = toUpload.getAbsolutePath();
-                File file = new File(fileName);
-                String mimetype = URLConnection.guessContentTypeFromName(file.getName());
-                InputStreamContent content = new InputStreamContent(mimetype, new FileInputStream(
-                        file));
-
-                picasaPhoto = client.executeInsertPhotoEntry(
-                        new PicasaUrl(albumToUse.getFeedLink()), content, toUpload.getName());
-
-                ContentValues cv = new ContentValues();
-                cv.put(Images.Media.PICASA_ID, picasaPhoto.id);
-
-                // update the content provider picasa_id once we upload
-                String where = Images.Media.DATA + "=?";
-                String[] whereArgs = {
-                        toUpload.getAbsolutePath()
-                };
-                Collect.getInstance().getContentResolver()
-                        .update(Images.Media.EXTERNAL_CONTENT_URI, cv, where, whereArgs);
-            }
+        //adding meta-data to the file
+        com.google.api.services.drive.model.File fileMetadata =
+                new com.google.api.services.drive.model.File()
+                        .setName(photosToUpload.get(key))
+                        .setViewersCanCopyContent(true)
+                        .setParents(Collections.singletonList(folder.getId()));
 
-            // uploadedPhotos keeps track of the uploaded URL
-            // relative to the path
-            uploaded.put(key, picasaPhoto);
-        }
+        FileContent mediaContent = new FileContent("image/jpeg", toUpload);
```

Comment:
I thought this would fail for pngs but in fact, it looks like PNGs are transferred and given a .jpg extension. That doesn't seem right!